### PR TITLE
[ADDED]Added validation for module.

### LINKED ISSUE #426: [Feat]: [Add Validation for Module]
## Description

There needs to be validation for all IDs in here. The validation must be done in the controller. You need to provide the service with a proper filter so that it can check if the ID provided in `req.body` actually exists for each collection that it belongs to. For example,
In `req.body` for adding Module, it needs to check that the provided IDs actually exist in the database. As always, never trust that the user is providing valid data.

## Use Case

It will make sure that data given by the user to the database is valid. 

## Proposed Solution

Right before the function that creates the document, check if the IDs provided by the user is valid.

### PR DESCRIPTION
**Pull request by deepti-50**
## Description
Used middleware validation service for validating ids and name of the module in controller/module.js file. 

## Fixes
Fixes #426 

## Checklist

- [x] Code follows project's style guidelines.
- [x] Changes are documented appropriately.

### CODE REVIEW

**Review by TejasNair9977 — APPROVED** [review]
LGTM

File: controller/module.js [hunk 1]
```diff
@@ -21,15 +23,22 @@ async function showModule(req, res) {
 
 async function addModule(req, res) {
   const { no, name, contents, hrsPerModule, cognitiveLevels } = req.body;
+  const isTopicValid = await isEntityIdValid(Topic, Topic);
```
Comment (sumitst05) [inline controller/module.js:26]: You are trying to compare `Topic` with `Topic` itself here... 
You should instead pass `contents` from `req.body`
For eg:
`isEntityValid(contents, Topic)` 
would be correct

File: controller/module.js [hunk 2]
```diff
@@ -21,15 +23,22 @@ async function showModule(req, res) {
...
+    if (isTopicValid) {
+      const newModule = await addNewModule(
+        no,
+        name,
+        contents,
+        hrsPerModule,
+        cognitiveLevels,
+      );
+      res.json({ res: `added module ${newModule.name}` });
+      res.json({ res: `added module ${newModule.id}` });
```
Comment (sumitst05) [inline controller/module.js:37]: can combine lines 36 and 37 in a single response

File: controller/module.js [hunk 3]
```diff
@@ -21,15 +23,22 @@ async function showModule(req, res) {
...
+    } else {
+      res.status(400).json({ err: "Invalid name" });
+      res.status(400).json({ err: "Invalid id" });
```
Comment (sumitst05) [inline controller/module.js:40]: same can be done here for 39 and 40

**Review by sumitst05 — CHANGES_REQUESTED** [review]
Please make the requested changes

**Comment by deepti-50** [comment]
@sumitst05 please see whether the changes are done right.

**Review by sumitst05 — APPROVED** [review]
LGTM!

**Pull request merged**